apps/web/src/components/activity/ExecutionDetailsModal.tsx

```diff
@@ -24,10 +23,14 @@ export function ExecutionDetailsModal({
 }) {
   const theme = useMantineTheme();
   const [shouldRefetch, setShouldRefetch] = useState(true);
-  const { data: response, isLoading } = useQuery(['activity', notificationId], () => getNotification(notificationId), {
-    enabled: !!notificationId,
-    refetchInterval: shouldRefetch ? 1000 : false,
-  });
+  const { data: response, isInitialLoading } = useQuery(
```

Comment:
another thing that was introduced `isInitialLoading`, which should be used when the query could be in the  [disabled state](https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#disabled-queries)